resources/src/ui/DetailsPanel.vue

```diff
@@ -458,14 +500,169 @@ async function onApply() {
+  stateSyncModal.reconcileMessage = '';
+}
+
+async function reconcileAndReapply() {
+  if (stateSyncModal.attemptingReconcile) {
+    return;
+  }
+
+  try {
+    stateSyncModal.attemptingReconcile = true;
+    stateSyncModal.reconcileMessage = '';
+
+    if (stateSyncModal.reconcileCommands.length === 0) {
+      stateSyncModal.reconcileMessage =
+        'No automatic reconciliation steps are available. Please sync with the backend instead.';
+      stateSyncModal.attemptingReconcile = false;
+      return;
+    }
+
+    for (const command of stateSyncModal.reconcileCommands) {
+      const response = await packsAction({
+        command: command.command,
+        repo_url: store.repoUrl,
+        ref: store.ref,
+        data: command.data,
+      });
+
+      if (!response.ok) {
+        throw new Error(response.message || `Failed to execute ${command.command}`);
+      }
+
+      if (response.diff) {
+        mergeDiff(store.packs, response.diff);
+      }
+      if (response.state_hash) {
+        store.stateHash = response.state_hash;
+      }
+      if (response.warnings) {
+        store.warnings = response.warnings;
+      }
+    }
+
+    stateSyncModal.reconcileMessage = 'Differences reapplied successfully. Attempting apply again...';
+    stateSyncModal.attemptingReconcile = false;
+    stateSyncModal.visible = false;
+
+    await onApply();
+  } catch (err) {
+    console.error('[reconcileAndReapply] Error:', err);
+    stateSyncModal.reconcileMessage = err instanceof Error ? err.message : String(err);
+    stateSyncModal.attemptingReconcile = false;
+  }
+}
+
+function deepClone(value) {
+  if (value === undefined) {
+    return value;
+  }
+  return JSON.parse(JSON.stringify(value));
+}
+
+function sanitizeDifferences(differences, commands = []) {
+  const result = {};
+  if (!differences || typeof differences !== 'object') {
+    return result;
+  }
+
+  const allowedPacks = new Set(
+    commands
+      .map((command) => command?.data?.pack_name)
+      .filter((packName) => typeof packName === 'string' && packName !== '')
+  );
+  const restrictToAllowed = allowedPacks.size > 0;
+
+  for (const [packName, packDiff] of Object.entries(differences)) {
+    if (restrictToAllowed && !allowedPacks.has(packName)) {
+      continue;
+    }
+
+    const fields = packDiff?.fields ?? {};
+    const pages = packDiff?.pages ?? {};
+
+    const fieldEntries = Object.entries(fields ?? {});
+
+    const filteredPages = {};
+    for (const [pageName, pageDiff] of Object.entries(pages ?? {})) {
+      const pageFields = Object.entries(pageDiff ?? {});
+      if (pageFields.length > 0) {
+        filteredPages[pageName] = pageDiff;
+      }
+    }
+
+    if (fieldEntries.length > 0 || Object.keys(filteredPages).length > 0) {
+      result[packName] = {
+        ...(fieldEntries.length > 0 ? { fields } : {}),
+        ...(Object.keys(filteredPages).length > 0 ? { pages: filteredPages } : {}),
+      };
+    }
+  }
+
+  return result;
+}
```

Comment:
why would we need a single purpose function for removing empty objects and arrays from an object when we are the ones creating the object? why not just construct the object correctly?